swarm/network/stream/messages.go

```diff
@@ -117,8 +117,7 @@ func (p *Peer) handleSubscribeMsg(ctx context.Context, req *SubscribeMsg) (err e
 
 	go func() {
 		if err := p.SendOfferedHashes(os, from, to); err != nil {
-			log.Warn("SendOfferedHashes dropping peer", "err", err)
-			p.Drop(err)
+			log.Warn("SendOfferedHashes error", "peer", p.ID().TerminalString(), "err", err)
```

Comment:
SendOfferedHashes returns an error if the priority queue is contended. Do you think the peer should be dropped because of this? But this can be a local problem with our node.

Reply:
yes i think it should be dropped. what else? we cannot go on

Reply:
If our network connection is not stable, we will drop peers randomly, although they are perfectly fine. I would at least retry a few times before dropping peer. Always immediately dropping can even make things worse in a non-perfect environment imho. We should at least measure the effect this on overall syncing performance